lib/cipherstash/protect/model/crud.rb

```diff
@@ -61,11 +75,47 @@ def protect_sync
           if search_attrs.kind_of?(Hash) && !search_attrs.empty?
             search_attrs.each do |virt_attr, metadata|
               searchable_attr = metadata[:searchable_attribute]
+              searchable_text_attr = metadata[:searchable_text_attribute]&.keys&.first
+
+              if searchable_attr
+                self.send("#{searchable_attr}=", self.send(virt_attr))
+              end
 
-              self.send("#{searchable_attr}=", self.send(virt_attr))
+              if searchable_text_attr
+                attr_value = self.send(virt_attr)
+                filter_options =  metadata[:searchable_text_attribute].fetch(searchable_text_attr)
+                bloom_filter_id = filter_options.fetch(:bloom_filter_id)
+
+                bits = CRUD.filter_bits(bloom_filter_id, filter_options, attr_value)
+
+                self.send("#{searchable_text_attr}=", bits)
+              end
             end
           end
         end
+
+        private
+
+        def self.filter_bits(bloom_filter_id, filter_options, value)
+          filter = CipherStash::Protect::ActiveRecordExtensions::BloomFilter.new(bloom_filter_id, { filter_size: filter_options[:filter_size], filter_term_bits: filter_options[:filter_term_bits] })
```

Comment:
As we noted in the huddle chat about this, we **could** cache this invocation by a combination of table + column (or _possibly_ model + column), but I would prefer to wait until we feel like that's needed for the sake of performance.